pkg/chartutil/values.go

```diff
@@ -376,3 +379,39 @@ func istable(v interface{}) bool {
 	_, ok := v.(map[string]interface{})
 	return ok
 }
+
+// PathValue takes a yaml path with . notation and returns the value if exists
+func (v Values) PathValue(ypath string) (interface{}, error) {
+	if len(ypath) == 0 {
+		return nil, error(fmt.Errorf("yaml path string cannot be zero length"))
+	}
+	yps := strings.Split(ypath, ".")
+	if len(yps) == 1 {
+		// if exists must be root key not table
+		vals := v.AsMap()
+		k := yps[0]
+		if _, ok := vals[k]; ok && !istable(vals[k]) {
+			// key found
+			return vals[yps[0]], nil
+		}
+		// key not found
+		return nil, ErrNoValue(fmt.Errorf("%v is not a value", k))
+	}
+	table := yps[:len(yps)-1]
```

Comment:
This is really hard to grok what is going on here until the end of the function. Could you comment it for more clarity? As I am reading this it almost seems like it is going down only one level (which isn't true). Probably more me misunderstanding this, but comments would probably be nice 😃